src/Avalonia.Base/Styling/StyleBase.cs

```diff
@@ -80,11 +82,46 @@ public bool TryGetResource(object key, out object? result)
             return _resources?.TryGetResource(key, out result) ?? false;
         }
 
-        internal void Attach(IStyleable target, IStyleActivator? activator)
+        internal ValueFrame Attach(IStyleable target, IStyleActivator? activator)
         {
-            var instance = new StyleInstance(this, target, _setters, _animations, activator);
-            target.StyleApplied(instance);
-            instance.Start();
+            if (target is not AvaloniaObject ao)
+                throw new InvalidOperationException("Styles can only be applied to AvaloniaObjects.");
```

Comment:
Due to GH formatting this almost looks like as it happens all the time. Maybe adding braces here wouln't hurt.

Reply:
Not sure what you mean here?